pkts-core/src/main/java/io/pkts/packet/IPPacket.java

```diff
@@ -129,27 +71,6 @@
      */
     int getTotalIPLength();
 
-    /**
-     * The checksum of the IP-packet. The checksum in an IP-packet is a 16 bit
-     * checksum of the header bytes (which the checksum set to zero) and is
-     * returned as a unsigned short (hence an int)
-     * 
-     * Checkout
-     * 
-     * @return
-     */
-    int getIpChecksum();
-
-    /**
-     * After you change anything in an IP packet (apart from the payload) you
-     * should re-calculate the checksum. If you don't, if this then is written
-     * to a pcap and later opened in e.g. wireshark, then all packets will be
-     * flagged as bad checksums.
-     */
-    void reCalculateChecksum();
-
-    boolean verifyIpChecksum();
```

Comment:
seems like this too would be true for IPv6

Reply:
IPv6 actually lacks any checksumming, leaving that up to layers above and below.

https://en.wikipedia.org/wiki/IPv6#Simplified_processing_by_routers

Reply:
ah, didn't know that, even found a post on stackoverflow regarding calculating checksum. Should have checked wikipedia first!